dash-pipeline/SAI/saithrift/Makefile

```diff
@@ -9,62 +9,57 @@ META=../SAI/meta
 
 SAIRPC_VENDOR_EXTRA_LIBS?=
 SAIRPC_EXTRA_LIBS= "$(SAIRPC_VENDOR_EXTRA_LIBS) \
-		-L/lib/x86_64-linux-gnu -Wl,-rpath=/lib/x86_64-linux-gnu -lm \
+		-L/lib/x86_64-linux-gnu -Wl,-rpath=/lib/x86_64-linux-gnu \
 		-L/usr/local/lib/ -Wl,-rpath=/usr/local/lib \
-	    -lpthread \
-	    -lpiprotogrpc \
-	    -lpiprotobuf \
-	    -lprotobuf \
-	    -lgrpc++ \
-	    -lgrpc \
-	    -lpiall \
-	    -lpi_dummy \
-	    -lpthread \
-	    -labsl_synchronization \
-	    -labsl_status \
-		-labsl_raw_hash_set \
-		-lgpr \
-		-lre2 \
-		-lssl \
-		-laddress_sorting"
+	    -lprotobuf -lpiprotobuf -lpiprotogrpc -lgrpc++"
+
+HUSER=hostuser
+HGROUP=hostgroup
+
+SUDO=sudo -u $(HUSER) -g $(HGROUP)
+
+host:
+	sudo groupadd -f -r -g $(HOST_GROUP) $(HGROUP)
+	sudo useradd $(HUSER) -l -u $(HOST_USER) -g $(HOST_GROUP) -d $(shell pwd) -m -s /usr/bin/bash
 
 # Below based on: https://github.com/opencomputeproject/SAI/blob/master/ptf/SAI_PTF_user-guide.md
 # Dependencies are assumed to be installed, e.g. in the docker container
-saithrift-server:
+saithrift-server: host
 	# Copy headers to /usr/include/sai
 	sudo mkdir -p /usr/include/sai
-	sudo cp $(SAI)/inc/sai*.h /usr/include/sai/
+	sudo cp -a $(SAI)/inc/sai*.h /usr/include/sai/
 	# Following is workaround for https://github.com/opencomputeproject/SAI/issues/1537
-	sudo cp $(SAI)/experimental/sai*.h /usr/include/sai/
-	sudo cp -r $(SAI)/experimental/ /usr/include/
+	sudo cp -a $(SAI)/experimental/sai*.h /usr/include/sai/
+	sudo cp -ar $(SAI)/experimental/ /usr/include/
 
 	# Install vendor specific SAI library i.e. DASH bmv2 libsai.so in /usr/lib.
 	sudo cp $(LIB)/libsai.so /usr/lib
 
 	@echo "Build SAI thrift server and libraries..."
-	mkdir -p $(RPC_INST_DIR) && chmod a+w $(RPC_INST_DIR)
+	$(SUDO) mkdir -p $(RPC_INST_DIR)
```

Comment:
This SUDO command and following ones - will that create root ownership of files/directories on the mounter host filesystem, and will it require interactive entering of passwords? We can't have that, we took pains to avoid it (hence the origin of the `chmod` commands).

Reply:
take a look at line 42/36 sudo is already there, and it will not ask passwords, since in dokcer files NO PASWORD is pushed to sudoers, also my solution will create files with uid/guid of host user, so there will be no conflict with repo to remove/modify, and touching files by chmod, so it speeds up process